setup-and-run.sh

```diff
@@ -416,7 +419,7 @@ ssl.truststore.type=JKS
 EOF
     sed -r -e "s|^(listeners=.*)|\1,SSL://:${BROKER_SSL_PORT}|" \
         -i /var/run/broker/server.properties
-    if [[ -n ${ADV_HOST} ]] && [[ -z ${KAFKA_ADVERTISED_LISTENERS} ]]; then
+    if [[ -n ${ADV_HOST} ]] && ${TMP_ADV_LISTENERS_SET}; then
```

Comment:
Whilst this is correct, let's replace it with our own idiom (used throughout the script) to make the script easier to read:
```
if [[ -n ${ADV_HOST} ]] && [[ $TMP_ADV_LISTENERS_SET =~ $TRUE_REG ]]; then
```